EasyPlantsApp/app/src/main/java/com/example/gamingpc/easyplants/MainActivity.java

```diff
@@ -78,31 +67,35 @@ protected void onCreate(Bundle savedInstanceState) {
 
         setup();
 
-        //Retrieve humidity value from firebase database
-        //TODO change path of child to read from message_list
-        DatabaseReference humidityValue = FirebaseDatabase.getInstance().getReference().child("HumidityTest").child("humidity_value");
-        //Query query = humidityValue.orderBy("name", Direction.DESCENDING).limit(3);
+        //Retrieve the last humidity value from updated from Firebase database message_list
+        DatabaseReference humidityValueRef = FirebaseDatabase.getInstance().getReference();
+        Query query = humidityValueRef.child("message_list").orderByKey().limitToLast(1);
```

Comment:
Hi, this line does not work. 

You are fetching the last data in the list. The list is sorted alphabetically. Thus fetching the last data will not fetch the latest data in all cases.

Reply:
I tested it and every time a new key is added it is given a key which has a higher alphabetical order. I tested it with more than 200 values and each key has a higher order. Thus, fetching the last data in the list will be the latest data in all cases.